sdk/identity/identity/src/credentials/usernamePasswordCredentialOptions.ts

```diff
@@ -1,9 +1,22 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { InteractiveCredentialOptions } from "./interactiveCredentialOptions";
+import { TokenCredentialOptions } from "../client/identityClient";
+import { TokenCachePersistenceOptions } from "../tokenCache/persistencePlatforms";
 
 /**
  * Defines options for the {@link UsernamePasswordCredential} class.
  */
-export interface UsernamePasswordCredentialOptions extends InteractiveCredentialOptions {}
+export interface UsernamePasswordCredentialOptions extends TokenCredentialOptions {
+  /**
+   * To provide a persistence layer to store the credentials,
+   * we allow users to optionally specify {@link TokenCachePersistenceOptions} for their credential.
+   *
+   * This feature is not currently available on Node 8 or earlier versions of Node JS.
+   *
+   * This persistence layer uses DPAPI on Windows.
+   * On OSX (Darwin) it tries to use the system's Keychain, otherwise if the property `allowUnencryptedStorage` is set to true, it uses an unencrypted file.
+   * On Linux it tries to use the system's Keyring, otherwise if the property `allowUnencryptedStorage` is set to true, it uses an unencrypted file.
```

Comment:
Is the file definitely unencrypted or just not protected by the native OS secret store?

Reply:
The file is unencrypted, only protected by file-system protections.

Here's one example in the source: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/identity/identity/src/tokenCache/persistencePlatforms.ts#L152-L155

MSAL for node offers a `FilePersistenceWithDataProtection` class, but it's only intended to be used on Windows.